src/ui.rs

```diff
@@ -0,0 +1,384 @@
+const GENERATIONS_POS_Y: f32 = 0.1;
+
+/// This function displays the user interface for the Game of Life, including rendering the game screen,
+/// handling user input, and switching between different modes (playing, paused, edit).
+///
+/// # Parameters
+///
+/// - `game`: The current game state.
+/// - `game_running`: A flag indicating whether the game is running.
+/// - `game_paused`: A flag indicating whether the game is paused.
+/// - `edit_mode`: A flag indicating whether the game is in edit mode.
+/// - `generations_passed`: The number of generations that have passed in the game.
+///
+async fn display_ui(
+    game: &mut Game,
+    game_paused: &mut bool,
+    edit_mode: &mut bool,
+    generations_passed: &mut i32,
+    acc: &mut f32,
+    fps: &mut f32,
+    prev_t: &mut Instant,
+) {
+    let elapsed = prev_t.elapsed().as_secs_f32();
+    *prev_t = Instant::now();
+
+    if !*game_paused {
+        *acc += elapsed;
+    }
+
+    handle_commands(game, game_paused, edit_mode, fps);
+
+    render_screen(game, generations_passed, game_paused, edit_mode, fps, acc);
+
+    next_frame().await;
+}
+
+/// Renders the game screen with grid lines and the live cells.
+///
+/// # Parameters
+///
+/// - `game`: The current game state.
+/// - `generations_passed`: The number of generations that have passed in the game.
+///
+fn render_screen(
+    game: &mut Game,
+    generations_passed: &mut i32,
+    game_paused: &mut bool,
+    edit_mode: &mut bool,
+    fps: &f32,
+    acc: &mut f32,
+) {
+    clear_background(LIGHTGRAY);
+    let screen_width = screen_width();
+    let screen_height = screen_height();
+
+    let game_size = screen_width.min(screen_height);
+    let offset_x = (screen_width - game_size) / 2. + 10.;
+    let offset_y = (screen_height - game_size) / 2. + 10.;
+    let sq_size = (screen_height - offset_y * 2.) / SQUARES as f32;
+
+    draw_rectangle(offset_x, offset_y, game_size - 20., game_size - 20., WHITE);
+
+    for i in 1..SQUARES {
+        draw_line(
+            offset_x,
+            offset_y + sq_size * i as f32,
+            screen_width - offset_x,
+            offset_y + sq_size * i as f32,
+            2.,
+            LIGHTGRAY,
+        );
+    }
+
+    for i in 1..SQUARES {
+        draw_line(
+            offset_x + sq_size * i as f32,
+            offset_y,
+            offset_x + sq_size * i as f32,
+            screen_height - offset_y,
+            2.,
+            LIGHTGRAY,
+        );
+    }
+
+    game.live_cells_do(|(x, y)| {
+        draw_rectangle(
+            offset_x + x as f32 * sq_size,
+            offset_y + y as f32 * sq_size,
+            sq_size,
+            sq_size,
+            GRAY,
+        )
+    });
+
+    let text = format!("Generations: {}", generations_passed);
+    draw_text(
+        &text,
+        screen_width * GENERATIONS_POS_X,
+        screen_height * GENERATIONS_POS_Y,
+        TEXT_SIZE,
+        BLACK,
+    );
+
+    let text = format!("FPS: {}", 1. / fps);
```

Comment:
A small suggestion for aesthetics:
```suggestion
    let text = format!("FPS: {}", (1. / fps).round());
```

Reply:
Done!